### ISSUE TITLE
Android - IOU-Scan via fab, little delayed display is noticed for category, tag and Billable

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#31241]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.98-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Launch app
2. Tap on employee WS
3. Tap plus icon--- Request money-Scan
4. Take a photo
5. Note category, tag(custom name-shahrukh khan) & Billable option are displayed instantly
6. Navigate to LHN
7. Tap fab
8. Tap request money---Scan
9. Take a photo
10. Select employee Workspace

## Expected Result:
While Scan via fab, the category, tag and Billable option must be instantly displayed

## Actual Result:
While Scan via fab, little delayed display is noticed for category, tag and Billable. But in WS scan, it's displayed instantly

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/711049c7-a3ea-40b3-8c80-0fe170c6baa0

https://github.com/Expensify/App/assets/78819774/54b3a7cf-7d45-4e14-9c57-d1fe36a2e50b

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by barttom** [comment]
Hi! I'm Bartek from Callstack and would like to work on this one! Will debug it and back with a proposal ASAP

**Comment by barttom** [comment]
I'm still debugging it.

**Comment by sonialiap** [comment]
Thanks Bartlomeij! Looking forwrad to it! I'm going to add you to the assignee list for easier tracking

**Comment by barttom** [comment]
I can't reproduce this issue. However, these are my thoughts.
In both cases we are navigated to the `src/pages/iou/steps/MoneyRequestConfirmPage.js`. 
This component manages to display items like Category, Tags etc. `src/components/MoneyRequestConfirmationList.js`

Here are the conditions that manage display tags and categories buttons
https://github.com/Expensify/App/blob/main/src/components/MoneyRequestConfirmationList.js#L223
https://github.com/Expensify/App/blob/main/src/components/MoneyRequestConfirmationList.js#L240

These conditions are based on Onyx values and simple find method `OptionsListUtils.hasEnabledOptions`
So, I can't see connections between the delay and places where the screen has been triggered from.
The suspect is that we fetch some data when we choose participants with FAB step, but need to check it.
Anyway, it's probably connected with connection speed or hardware. 


https://github.com/Expensify/App/assets/7682108/fa17815f-c580-45cf-a05e-241f8634a458

**Comment by sonialiap** [comment]
@robertKozik what do you think of Bartlomeij's thoughts in the above comment?

**Comment by sonialiap** [comment]
@robertKozik bumping :)

**Comment by robertKozik** [comment]
I see that you checked in on fresh account @barttom . Did you test on the hightraffic account as well, as we are accessing the page which could load lots of personalDetails?
I can't reproduce it on my end as well, but I just want to be sure that we would not unnecessary close it.

**Comment by barttom** [comment]
Yes, I've checked it also on another account and can't reproduce it.
